graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -718,6 +719,21 @@ def _compile_ast_node_to_ir(schema, current_schema_type, ast, location, context)
     return basic_blocks
 
 
+def _validate_all_tags_are_used(metadata):
+    """Ensure all tags are used in some filter."""
+    tag_names = set(metadata.tags)
+    filter_args = set()
+    for location, _ in metadata.registered_locations:
+        for filter_info in metadata.get_filter_infos(location):
+            for arg in filter_info.args:
+                if _is_tag_argument(arg):
+                    filter_args.add(arg[1:])  # Don't include % in tag name
```

Comment:
Let's maybe also add a helper for this, in the same place where `_is_tag_argument()` is getting moved to.

Reply:
Also can you explain what args mean? I think Predrag already mentioned a way to address this issue before.